opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -18,9 +18,16 @@
 import uuid
 from functools import wraps
 from os import environ
+from sys import version_info
 
-from pkg_resources import iter_entry_points
+# FIXME remove when support for 3.7 is dropped.
+if version_info.minor == 7:
+    # pylint: disable=import-error
+    from importlib_metadata import entry_points  # type: ignore
+else:
+    from importlib.metadata import entry_points
```

Comment:
i'm not sure but i think that importlib does not cache entry points and reads them every time `entry_points` is called.
when i checked it, it took about 5-10 ms for each call to `entry_points`. since it probably depends on the number of packages installed these times are not representative but indicate that there is some overhead.

afaik `pkg_resources` (although much slower than importlib) caches everything when the module is first imported, so only the first import of the module takes some time.

since OTel uses quite a number of entry points maybe it would make sense to lazily load and cache them?

Reply:
Opened #3054 to track this, this PR is already quite big.